File: valo_tui/style/icons.py
"""Single-character semantic icons for maps and agent roles.

Deliberately tiny — one glyph + colour, never giant art. The role colours
match the palette accents so a scoreboard reads at a glance.
"""

from __future__ import annotations

# ── Maps ─────────────────────────────────────────────────────
MAP_ICONS: dict[str, str] = {
    "Ascent": "◆",
    "Bind": "◈",
    "Haven": "▲",
    "Lotus": "❀",
    "Sunset": "☀",
    "Split": "║",
    "Icebox": "❄",
    "Pearl": "○",
    "Breeze": "≈",
    "Fracture": "⚡",
    "Abyss": "▽",
    "Corrode": "◙",
}


def map_icon(name: str | None) -> str:
    return MAP_ICONS.get((name or "").strip().title(), "·")


# ── Agent roles ──────────────────────────────────────────────
# role -> (glyph, colour)
ROLE_GLYPH: dict[str, tuple[str, str]] = {
    "duelist": ("▲", "#e0594a"),
    "controller": ("◆", "#5d9ce8"),
    "initiator": ("◈", "#e8c15d"),
    "sentinel": ("●", "#5de88a"),
}

_AGENT_ROLE: dict[str, str] = {}
for _role, _agents in {
    "duelist": "jett raze phoenix reyna yoru neon iso waylay",
    "controller": "brimstone omen viper astra harbor clove",
    "initiator": "sova breach skye kayo fade gekko tejo",
    "sentinel": "killjoy cypher sage chamber deadlock vyse",
}.items():
    for _a in _agents.split():
        _AGENT_ROLE[_a] = _role


def agent_role(name: str | None) -> str | None:
    return _AGENT_ROLE.get((name or "").strip().lower())


def agent_glyph(name: str | None) -> tuple[str, str]:
    """Return ``(glyph, colour)`` for an agent, defaulting to a muted dot."""
    return ROLE_GLYPH.get(agent_role(name) or "", ("·", "#4a708b"))

File: internal/styles/icons.go
package styles

import (
	"image/color"
	"strings"
)

// Single-character semantic icons for maps and agent roles, ported from
// style/icons.py. Deliberately tiny — one glyph + colour.

var mapIcons = map[string]string{
	"Ascent": "◆", "Bind": "◈", "Haven": "▲", "Lotus": "❀",
	"Sunset": "☀", "Split": "║", "Icebox": "❄", "Pearl": "○",
	"Breeze": "≈", "Fracture": "⚡", "Abyss": "▽", "Corrode": "◙",
}

// MapIcon returns the glyph for a map name, or "·" if unknown.
func MapIcon(name string) string {
	if g, ok := mapIcons[titleCase(strings.TrimSpace(name))]; ok {
		return g
	}
	return "·"
}

// titleCase upper-cases the first rune and lower-cases the rest (map names are
// single words), matching Python str.title() for this use.
func titleCase(s string) string {
	if s == "" {
		return ""
	}
	r := []rune(strings.ToLower(s))
	r[0] = []rune(strings.ToUpper(string(r[0])))[0]
	return string(r)
}

type roleStyle struct {
	glyph  string
	colour color.Color
}

var roleGlyph = map[string]roleStyle{
	"duelist":    {"▲", RoleDuelist},
	"controller": {"◆", RoleController},
	"initiator":  {"◈", RoleInitiator},
	"sentinel":   {"●", RoleSentinel},
}

var agentRole = func() map[string]string {
	m := map[string]string{}
	for role, agents := range map[string]string{
		"duelist":    "jett raze phoenix reyna yoru neon iso waylay",
		"controller": "brimstone omen viper astra harbor clove",
		"initiator":  "sova breach skye kayo fade gekko tejo",
		"sentinel":   "killjoy cypher sage chamber deadlock vyse",
	} {
		for _, a := range strings.Fields(agents) {
			m[a] = role
		}
	}
	return m
}()

// AgentRole returns an agent's role, or "" if unknown.
func AgentRole(name string) string {
	return agentRole[strings.ToLower(strings.TrimSpace(name))]
}

// AgentGlyph returns (glyph, colour) for an agent, defaulting to a muted dot.
func AgentGlyph(name string) (string, color.Color) {
	if rs, ok := roleGlyph[AgentRole(name)]; ok {
		return rs.glyph, rs.colour
	}
	return "·", Muted
}
